server/src/main/java/org/elasticsearch/action/admin/cluster/snapshots/get/TransportGetSnapshotsAction.java

```diff
@@ -540,13 +559,81 @@ private static SnapshotsInRepo sortAndFilterSnapshots(
             final Predicate<SnapshotInfo> fromSortValuePredicate = buildFromSortValuePredicate(sortBy, fromSortValue, order, null, null);
             if (predicate == null) {
                 predicate = fromSortValuePredicate;
-            } else {
+            } else if (fromSortValuePredicate != null) {
                 predicate = fromSortValuePredicate.and(predicate);
             }
         }
         return predicate;
     }
 
+    // Builds a predicate that can be applied to a combination of snapshot id and repository data to filter out snapshots that are not
```

Comment:
Regular Javadoc blocks look better in most tools/IDE, why not use them?

Reply:
++ moved to regular docs all around :)